app/models/manageiq/providers/redhat/infra_manager/vm/operations/snapshot.rb

```diff
@@ -1,4 +1,18 @@
 module ManageIQ::Providers::Redhat::InfraManager::Vm::Operations::Snapshot
+  extend ActiveSupport::Concern
+
+  included do
+    supports :snapshots do
+      unless ext_management_system&.supports_snapshots?
+        if ext_management_system
```

Comment:
I have a similar logic in my PR:
https://github.com/ManageIQ/manageiq/pull/11971/files#diff-ea4cf39a1d54c20ad7a08131c41cbea0R12

```
supports 'feature-name' do
   if provider does not exist
       add message
   else if provider does not support 'feature-name' 
       add message
```
Can we extract it to be reused ?